### ISSUE TITLE
Automate the placement of IfcCovering on IfcWall and IfcCeiling

### ISSUE THREAD

**Issue by Owuraqu** [IfcOpenShell/IfcOpenShell#3754]
There should be the functionality to help automate IfcCovering on IfcWall and IfcSlab. I searched BBIM, ifcopenshell api and Ifcsverchok and haven’t found anything yet. 

For example in RevitAPI and Dynamo I am able to easily extract the boundary lines of Room and use it as footprint for creation and placement wall and ceiling coverings/ finishes. 

Similar automation process should be made available with BBIM and ifcopenshell api or Ifcsverchok. This would be a huge time saver for true BIM modelling.

**Comment by maxfb87** [comment]
I can try to do something about it :-)
I guess it should work in a similar way of create spaces from cursor position and from walls so i can try to do something...
I'll write here for updates :-) of course help is welcome

**Comment by Owuraqu** [comment]
That would be great and thank you very much 😊

**Comment by blenderarc** [comment]
Can this be used? @maxfb87

**Comment by maxfb87** [comment]
@blenderarc yep, not finished, but now it's possible to create a flooring or ceiling from selected walls or from cursor position.
To do that, you have to create first the type (IfcCoveringType) with predefined type "FLOORING" or "CEILING".
After that, select the Covering Tool, select the type you want from the n-panel and you should be able to create the objects...
Waiting for feedback :-)

**Comment by blenderarc** [comment]
![Image](https://github.com/IfcOpenShell/IfcOpenShell/assets/140309875/e0ece16a-624c-4edb-bd4d-5461ff568654)



![Image](https://github.com/IfcOpenShell/IfcOpenShell/assets/140309875/4fc420ba-fc69-4820-b227-063efe1aa1c0)

**Comment by Owuraqu** [comment]
Awesome @maxfb87 ☺️. Now BIMing now simplified! Will try it up and thank you very much for the effort.

**Comment by maxfb87** [comment]
@blenderarc i'm not sure if i understood correctly your pics, have you managed to use the tool?

@Owuraqu thanks :-) looking forward to continue..

**Comment by blenderarc** [comment]
NOT SUCCESSFUL

**Comment by blenderarc** [comment]
not successful

**Comment by maxfb87** [comment]
@blenderarc could you share the ifc file? So i can help you...

**Comment by blenderarc** [comment]
[COVERING CEILING.zip](https://github.com/IfcOpenShell/IfcOpenShell/files/13316927/COVERING.CEILING.zip)

**Comment by maxfb87** [comment]
@blenderarc thanks...well first of all in order to create the flooring/ceiling from cursor position it is necessary to select the storey collection object..this is why you don't see the operator in the UI.
Also, i found a bug if the project units are in millimetres instead of metres..
I'm trying to fix it

**Comment by Dayo9174** [comment]
hi, is there any way we can also do this for walls? I am trying to add insulation that runs the perimeter of a foundation wall. 

Thanks in advance for your help.

**Comment by maxfb87** [comment]
@Dayo9174 yeah, i had in mind also that but atm i don't exactly know how to do it..

**Comment by Owuraqu** [comment]
> @Dayo9174 yeah, i had in mind also that but atm i don't exactly know how to do it..
I don't know if the wall side area and non-parallel intersection walls could be useful.

In this case for the height of the wall covering the user specifies the height offset needed from the extracted wall side area.

And for the horizontal extents the trimmed intersection walls becomes the boundary ends for the covering.

Just my kindergarten thoughts:).

**Comment by Owuraqu** [comment]
Hi @maxfb87 I had an idea for the automation of the wall, floor and ceiling covering. I don't know if it will be practically possible but what about using IfcSpace. 

Considering that the IfcSpace (cube) is bounded by walls, floors, and a ceiling can't it be used to automate the creation of IfcCovering for walls, floors and ceiling? Of course I am thinking of simple spaces with not much complexities.

There should be a way to create this so kindly let me know your thoughts on this. Regards:)

**Comment by maxfb87** [comment]
Hi @Owuraqu actually creating flooring and ceiling with the covering tool works using the same functions that "create space" uses so your intuition is right :-)
The problem i had was about create coverings like cladding from walls.
Unfortunately i don't have much time to spend on developing but i wrote a script that could be a good start i could share it if anybody is interested
